airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -305,3 +306,43 @@ definitions:
           oauth2Specification:
             description: If the connector supports OAuth, this field should be non-null.
             "$ref": "#/definitions/OAuth2Specification"
+      advanced_auth:
+        type: object
+        properties:
+          auth_flow_type:
+            type: string
+            enum: ["oauth2.0"] # Future auth types should be added here
+          predicate_key:
+            description: Json Path to a field in the connectorSpecification that should exist for the advanced auth to be applicable.
+            type: array
+            items:
+              type: string
+          predicate_value:
+            description: Value of the predicate_key fields for the advanced auth to be applicable.
+            type: string
+          oauth2_specification:
+            "$ref": "#/definitions/OAuth2ConfigSpecification"
+  OAuth2ConfigSpecification:
+    type: object
+    properties:
+      input_specification:
+        description: "OAuth2 specific blob.
+          Must be a valid non-nested JSON that refers to properties from ConnectorSpecification.connectionSpecification
+          using special annotation 'path_in_connector_config'."
+        type: object
+        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+      parameter_specification:
```

Comment:
can we call this something like `complete_oauth_flow_input_specification` to be clear about the part of the process it plays?